homeassistant/components/seventeentrack/sensor.py

```diff
@@ -111,242 +95,150 @@ async def async_setup_entry(
-
-        package_data = []
-        for package in self._data.packages.values():
-            if package.status != self._status:
-                continue
-
-            package_data.append(
-                {
-                    ATTR_FRIENDLY_NAME: package.friendly_name,
-                    ATTR_INFO_TEXT: package.info_text,
-                    ATTR_TIMESTAMP: package.timestamp,
-                    ATTR_STATUS: package.status,
-                    ATTR_LOCATION: package.location,
-                    ATTR_TRACKING_NUMBER: package.tracking_number,
-                }
-            )
-
-        self._attr_extra_state_attributes[ATTR_PACKAGES] = (
-            package_data if package_data else None
-        )
+        """Return the state of the sensor."""
+        return self.coordinator.data.summary[self._status]["quantity"]
 
-        self._state = self._data.summary.get(self._status)
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return the state attributes."""
+        return {ATTR_PACKAGES: self.coordinator.data.summary[self._status]["packages"]}
 
 
-class SeventeenTrackPackageSensor(SensorEntity):
+class SeventeenTrackPackageSensor(
+    CoordinatorEntity[SeventeenTrackCoordinator], SensorEntity
+):
     """Define an individual package sensor."""
 
     _attr_attribution = ATTRIBUTION
     _attr_icon = "mdi:package"
 
-    def __init__(self, data, package) -> None:
-        """Initialize."""
-        self._attr_extra_state_attributes = {
-            ATTR_DESTINATION_COUNTRY: package.destination_country,
-            ATTR_INFO_TEXT: package.info_text,
-            ATTR_TIMESTAMP: package.timestamp,
-            ATTR_LOCATION: package.location,
-            ATTR_ORIGIN_COUNTRY: package.origin_country,
-            ATTR_PACKAGE_TYPE: package.package_type,
-            ATTR_TRACKING_INFO_LANGUAGE: package.tracking_info_language,
-            ATTR_TRACKING_NUMBER: package.tracking_number,
-        }
-        self._data = data
-        self._friendly_name = package.friendly_name
-        self._state = package.status
-        self._tracking_number = package.tracking_number
-        self.entity_id = ENTITY_ID_TEMPLATE.format(self._tracking_number)
+    def __init__(
+        self,
+        coordinator: SeventeenTrackCoordinator,
+        tracking_number: str,
+        friendly_name: str | None = None,
+        status: str | None = None,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._tracking_number = tracking_number
+        self._status = status
+        self.entity_id = ENTITY_ID_TEMPLATE.format(tracking_number)
         self._attr_unique_id = UNIQUE_ID_TEMPLATE.format(
-            data.account_id, self._tracking_number
+            coordinator.account_id, tracking_number
         )
+        name = friendly_name if friendly_name else tracking_number
+        self._name = f"Seventeentrack Package: {name}"
```

Comment:
Why did we move this again?

Reply:
there is no need to calculate this each time property name is called.
We have the information at __init__ time